### PR TITLE
Lt 4280 update rabbit mq libraries

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/MarginTrading.Backend.Services/Workflow/PlatformClosureProjection.cs [hunk 1]
```diff
@@ -32,19 +39,41 @@ public async Task Handle(MarketStateChangedEvent e)
...
+                    SnapshotStatus.Draft);
+            }
+            catch (Exception ex)
+            {
+                await _log.WriteWarningAsync(nameof(PlatformClosureProjection),
+                    nameof(Handle),
+                    e.ToJson(),
+                    $"Failed to make trading data draft snapshot for [{tradingDay}]", ex);
+                
+                if (tradingDay < _dateService.NowDateOnly())
```
Comment (fatihdumanli) [inline src/MarginTrading.Backend.Services/Workflow/PlatformClosureProjection.cs:58]: Probably I don't have the full facts but, can't we make this check without the exception being thrown?  
If it's not a valid state, couldn't it have been checked before the attempt to create the snapshot?

File: src/MarginTrading.Backend/Middleware/GlobalErrorHandlerMiddleware.cs [hunk 2]
```diff
@@ -43,28 +46,33 @@ public async Task Invoke(HttpContext context)
                     return;
                 }
 
-                await Log(ex, asInfo: ex is LogInfoOnlyException);
+
```
Comment (fatihdumanli) [inline src/MarginTrading.Backend/Middleware/GlobalErrorHandlerMiddleware.cs:49]: Three blank lines

File: src/MarginTrading.Common/Helpers/StreamHelpers.cs [hunk 3]
```diff
@@ -1,23 +1,79 @@
...
     {
-        public static async Task<string> GetStreamPart(Stream stream, int maxPart)
+        /// <summary>
+        /// Copies the stream and reads a part of it. Assumes that the stream is seekable.
+        /// Changes the source stream position to the beginning of the stream.
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <param name="maxBytes"></param>
+        /// <returns></returns>
+        public static async Task<byte[]> ReadBytes(this Stream stream, int maxBytes)
```
Comment (fatihdumanli) [inline src/MarginTrading.Common/Helpers/StreamHelpers.cs:19]: `MaxBytes` arg can be `uint` and that eliminates the if block starts at line 38.

File: src/MarginTrading.Common/Helpers/StreamHelpers.cs [hunk 4]
```diff
@@ -1,23 +1,79 @@
...
+
+                ms.Seek(0, SeekOrigin.Begin);
 
-            var reader = new StreamReader(stream);
-            var len = (int)(stream.Length > maxPart ? maxPart : stream.Length);
-            var bodyPart = new char[len];
-            await reader.ReadAsync(bodyPart, 0, len);
+                int bytesToRead = Math.Min((int)ms.Length, maxBytes);
+                int bytesRead = 0;
+                byte[] buffer = new byte[bytesToRead];
```
Comment (fatihdumanli) [inline src/MarginTrading.Common/Helpers/StreamHelpers.cs:63]: If we have a buffer with a fixed size (which is evaluated as `Math.Min(ms.Length, maxBytes)` before reading the stream, why do we need a loop to read the stream? 

In each iteration, it reads X bytes, where X is equal to the length of the buffer. So after one iteration, the buffer should be full. There won't be second execution. So doesn't that eliminates the need for a loop?

Am I missing something?

File: src/MarginTrading.Common/Middleware/RequestsLoggingMiddleware.cs [hunk 5]
```diff
@@ -55,7 +55,8 @@ public async Task Invoke(HttpContext context)
 
                     using (originalRequestBody)
                     {
-                        var body = await StreamHelpers.GetStreamPart(originalRequestBody, _settings.MaxPartSize);
+                        var bytes = await originalRequestBody.ReadBytes(_settings.MaxPartSize);
```
Comment (fatihdumanli) [inline src/MarginTrading.Common/Middleware/RequestsLoggingMiddleware.cs:58]: Same, do we need to log the body if it's null?

File: src/MarginTrading.Backend.Services/Workflow/PlatformClosureProjection.cs (hunk 1 above)
Comment (tarurar) [reply src/MarginTrading.Backend.Services/Workflow/PlatformClosureProjection.cs:58]: Making draft snapshot for the past. for example, yesterday, is not a problem. 
This code is intended catch the case when we have  accumulated a lot of MarketStateChangedEvent for the past (if rabbit mq was down for a long time, for example, it usually happens on dev env).

File: src/MarginTrading.Backend/Middleware/GlobalErrorHandlerMiddleware.cs [hunk 6]
```diff
@@ -43,28 +46,33 @@ public async Task Invoke(HttpContext context)
...
-        private async Task Log(Exception ex, bool asInfo)
+        private async Task LogWithRequest(HttpRequest request, Exception ex, bool asInfo)
         {
-            string bodyPart;
-
-            using (var ms = new MemoryStream())
-            {
-                bodyPart = await StreamHelpers.GetStreamPart(ms, 1024);
-            }
+            var bytes = await request.Body.ReadBytes(_settings.MaxPartSize);
```
[the remark this answers is not in the document]
Comment (tarurar) [reply src/MarginTrading.Backend/Middleware/GlobalErrorHandlerMiddleware.cs:59]: _settings.MaxPartSize is an old configuration key which was introduces, but wasn't used here. I fixed that.

File: src/MarginTrading.Backend/Middleware/GlobalErrorHandlerMiddleware.cs [hunk 7]
```diff
@@ -43,28 +46,33 @@ public async Task Invoke(HttpContext context)
...
+            var bodyPart = bytes == null ? null : System.Text.Encoding.UTF8.GetString(bytes);
```
[the remark this answers is not in the document]
Comment (tarurar) [reply src/MarginTrading.Backend/Middleware/GlobalErrorHandlerMiddleware.cs:60]: We also log information about the exception.

File: src/MarginTrading.Common/Helpers/StreamHelpers.cs (hunk 4 above)
Comment (tarurar) [reply src/MarginTrading.Common/Helpers/StreamHelpers.cs:63]: Not exactly, at each iteration it reads at maximum X bytes. It is asynchronous stream reading, we express a desire to read at maximum bytesToRead but if, by any chance we read less (ReadAsync returns the actual bytes read) then we iterate.

**Pull request merged**